TrafficCapture/nettyWireLogging/src/test/java/org/opensearch/migrations/trafficcapture/netty/ConditionallyReliableLoggingHttpRequestHandlerTest.java

```diff
@@ -66,14 +83,75 @@ private static void writeMessageAndVerify(byte[] fullTrafficBytes, Consumer<Embe
         Assertions.assertEquals(1, flushCount.get());
     }
 
-    @Test
-    public void testThatTinyPacketsPostBlocks() throws IOException {
+    private static byte[] consumeIntoArray(ByteBuf m) {
+        var bArr = new byte[m.readableBytes()];
+        m.readBytes(bArr);
+        m.release();
+        return bArr;
+    }
+
+    private ByteBuf getByteBuf(byte[] src, boolean usePool) {
+        var unpooled = Unpooled.wrappedBuffer(src);
+        if (usePool) {
+            var pooled = ByteBufAllocator.DEFAULT.buffer(src.length);
+            pooled.writeBytes(unpooled);
+            unpooled.release();
+            return pooled;
+        } else {
+            return unpooled;
+        }
+    }
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testThatAPostInASinglePacketBlocksFutureActivity(boolean usePool) throws IOException {
+        byte[] fullTrafficBytes = SimpleRequests.SMALL_POST.getBytes(StandardCharsets.UTF_8);
+        var bb = getByteBuf(fullTrafficBytes, usePool);
+        writeMessageAndVerify(fullTrafficBytes, w -> {
+            w.writeInbound(bb);
+        });
+        log.info("buf.refCnt="+bb.refCnt());
+        //Assertions.assertEquals(0, bigBuff.refCnt());
```

Comment:
Why is this commented out? It's essentially validating that the ref counting is working, right?

Reply:
Yes.  I've deleted the line in the next commit.  It should be getting subsumed by the testThatAPostInASinglePacketBlocksFutureActivity_withLeakDetection test.  I figured that I'd leave this test to the logic OTHER than memory leaks & handle all leak stuff separately.